packages/vaadin-grid/test/keyboard-navigation.test.js

```diff
@@ -1956,17 +1956,47 @@ describe('keyboard navigation', () => {
 
       const spy = sinon.spy();
 
-      grid.addEventListener('cell-focus', (e) => {
+      const eventHandler = (e) => {
         spy();
 
         const context = e.target.getEventContext(e);
         expect(context).to.deep.equal(expectedContext);
         done();
-      });
+      };
+
+      grid.addEventListener('cell-focus', eventHandler);
 
       left();
 
       expect(spy.calledOnce).to.be.true;
+      grid.removeEventListener('cell-focus', eventHandler);
+    });
+
+    it('should contain event context in event detail', () => {
+      tabToBody();
+      right();
+
+      const spy = sinon.spy();
+
+      grid.addEventListener('cell-focus', spy);
+
+      left();
+
+      const e = spy.firstCall.args[0];
+
+      expect(e.detail.context).to.be.deep.equal({
+        column: grid.querySelector('vaadin-grid-column'),
```

Comment:
Note that it will also deeply compare every property of `vaadin-grid-column`, which might not be efficient. (it is how [deep-eql](https://github.com/chaijs/deep-eql) works)

In this case, using strict equality may be a better choice.

```js
expect(e.detail).to.be.ok;
expect(e.detail.context).to.be.ok;
expect(e.detail.context.column).to.equal(grid.querySelector('vaadin-grid-column')); // <---------
```